joomla/joomla-cms
# Fix merge priority for contact item params

### Summary of Changes
The parameter cascade / priority for almost all(?) entities in Joomla! works in the same way. In short, the order is (should be) as follows: menu item > entity > category > component / global.

This seems not to be the case for com_contact, where a contact item, even if it was assigned to a menu item, overrules the menu items parameters.

### Testing Instructions
Create a contact item, assign a user to it and / or give it a "valid" email address. Also make sure to set the "Contact Form" option in the "Form" tab of that view to "Show".

Next, create a menu item for the single contact and set "Contact Form" in the "Mail Options" tab to "Hide"

Now head over to the com_contact options and click the set "Form" tab. There, set "Contact Form" in the "Form" tab to "Hide".

In Short:
"Contact Form" Option
- component (global): "Hide"
- contact item: "Show"
- menu item: "Hide"

### Expected result
When you call the contact via the created link, the email form should be hidden.

### Actual result
The email form still shows up.

### Documentation Changes Required
N/A

### Caveat
While fixing the merge priority for the contact item params, I've tried to add as few lines as possible and as much as needed. The focus of this PR was to fix the behavior (bug) where a single contact item (linked) overrules the menu settings, which it should not.

If there is a bug that shows in another scenario, e.g. where either a category of contacts or [you name it] does not work as expected*, it will still be broken even after this patch and must be addressed in a different / new PR. From what I grasped while strolling through the code, there is a good chance that a few other things are broken in com_contact.

That said, apart from the default layout, I've also changed the value of the custom fields layout params variable because it seemed to be affected by the bug as well.

*adhere to the params cascade / priority

Big thanks to Benjamin for lending me a second pair of eyes!
